### ISSUE TITLE
'Allow Interruptions' does not work when set to true for Multi-Choice Prompts 

### ISSUE THREAD

**Issue by mn-invia** [microsoft/BotFramework-Composer#7303]
Hi There,

I have a Chatbot which has been built using Microsoft Composer. At several instances, I am providing the user Multi-Choice prompts to select/choose options depending upon the scenario. I have set the 'Allow Interruptions' Flag to 'TRUE'. I am trying to initiate small talk when these select/choose options are provided to the user and it is NOT WORKING. The Multi-Choice Prompts keep appearing time and again. It does not allow for any 'Small-Talk' when the Multi-Choice Prompts are used.....but the 'Small-Talk' does work when it's a Single choice prompt (any text or number or date input).

OS: Windows 10 Pro
Browser: Google Chrome
Composer:
Version:  1.4.0
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

Please let me know as to what more you need from me to get this issue resolved. Many Thanks & Best Regards.

**Comment by stevkan** [comment]
@mn-invia, thank you for your patience while I looked into this. When I attempt a repro I am able to setup a multi-choice prompt, set the 'Allow Interruptions' flag to `true`, and it works fine for me. If I type in a response that is not on the list, I am reprompted. If I type in 'cancel', then the dialog is cancelled and I am returned to the parent dialog.

Two questions:

What does your cancel trigger and dialog look like?
And, what do you mean by 'Small-Talk'? Are you referring to QnA's chit-chat knowledgebase option? And, how is the user selecting an option tied to this?

**Comment by mn-invia** [comment]
Hey Steven,

Many Thanks for that.

Yes….I also have the same scenario, When I also type in a response that is not in the list….It re-prompts me promptly. What I needed was when the user is given a Multi-choice prompt and instead of selecting one of the options provided, if he tries to have ‘small-talk’ chat with the bot….it does not recognize it. For example…..the user instead of choosing one of the options, he types in ‘Hello, how are you?’….it should kick off the QnA which I have already set-up. The QnA works well when we bot prompts for user input (single input)….but does not work when provided with Multi-Choice inputs. I hope I have given you enough inputs…..Probably it might be currently the limitation of the Microsoft composer. Let me know if you need more information. Once again Many Thanks for that.

Best Regards,
Murali

**Comment by stevkan** [comment]
@mn-invia, thank you for your patience while I looked into this. I tried again and I am able to interrupt the multi-choice prompt with a 'QnA Intent recognized' trigger by asking a question that the QnA resource can respond to. It's important to note that when you use the QnA trigger, the trigger is accessing a locally resourced knowledgebase, when run locally. The knowledgebase is designed from scratch or uploaded from a file within the QnA form in Composer (see images below). It _does not_ access QnA resources from [www.qnamaker.ai](https://www.qnamaker.ai) when run locally, but will publish the KB to QnAMaker.ai as the Composer bot is published.

If you are expecting the QnA trigger to respond by including only the QnA credentials in the Composer Settings, this will not work. The bot will ignore the trigger as it hasn't been correctly configured (i.e. a local KB having been setup). The bot settings are only referenced when using the Access external resources >> Connect to QnA Knowledgebase action within a dialog.

![image](https://user-images.githubusercontent.com/18403946/118094176-4256bb00-b383-11eb-9a0e-b2a50c028230.png)

![image](https://user-images.githubusercontent.com/18403946/118094191-4682d880-b383-11eb-9774-e417f0a623ca.png)
